calculator.py

```diff
@@ -58,17 +58,24 @@ def checkOperators(expList):
 		if((expression[index-1] in "+-") and (expression[index] in "*/")): return False
 		#Operators and parenthesis
 		if(((expression[index-1]=="(") and (expression[index] in "*/")) or ((expression[index-1] in "*/+-") and (expression[index]==")"))): 
-			return False 
+			return False 			
 	expList[0] = expression
 	return True
 
 def checkDecimalPoint(expList):
 	expression = expList[0]
 	index = 0
-	if(expression[0]=="." or expression[-1]=="."): return False
-	for char in expression: #If there
-		if((expression[index]==".") and (index != 0) and (index != len(expression)-1) and (not(expression[index-1] in "0123456789" and expression[index+1] in "0123456789"))):
-			return False
+	validPointApperance="false";
+	#We make sure that, once received a point, we do not receive more on the same number
+	for char in expression:

```

Comment:
It doesn't catch when the decimal point is at the beginning or at the end of an expression, nor when it is right before or after an operator.

Reply:
Neither does the original version if the number is not isolated, such as 3.+2, or if there exist an operator before the point, such as -.3+2. That's the reason I thought it was expected behavior.
A solution is to merge the new code without deleting the old function, getting rid of the bug I solved but keeping the one described above. I'll update the files and pull if it's OK for you